src/js/bandwidth_test.js

```diff
@@ -227,75 +227,95 @@ VideoBandwidthTest.prototype = {
 };
 
 addExplicitTest(testSuiteName.THROUGHPUT, testCaseName.NETWORKLATENCY,
-  Call.asyncCreateTurnConfig.bind(null, wiFiPeriodicScanTest.bind(null,
-      Call.isNotHostCandidate), reportFatal));
+  function(test) {
+    var wiFiPeriodicScanTest = new WiFiPeriodicScanTest(test);
+    wiFiPeriodicScanTest.run(Call.isNotHostCandidate);
+  });
 
 addExplicitTest(testSuiteName.THROUGHPUT, testCaseName.NETWORKLATENCYRELAY,
-  Call.asyncCreateTurnConfig.bind(null, wiFiPeriodicScanTest.bind(null,
-      Call.isRelay), reportFatal));
-
-function wiFiPeriodicScanTest(candidateFilter, config) {
-  var testDurationMs = 5 * 60 * 1000;
-  var sendIntervalMs = 100;
-  var running = true;
-  var delays = [];
-  var recvTimeStamps = [];
-  var call = new Call(config);
-  var chart = createLineChart();
-  call.setIceCandidateFilter(candidateFilter);
-
-  var senderChannel = call.pc1.createDataChannel({ordered: false,
-                                                  maxRetransmits: 0});
-  senderChannel.addEventListener('open', send);
-  call.pc2.addEventListener('datachannel', onReceiverChannel);
-  call.establishConnection();
-
-  setTimeoutWithProgressBar(finishTest, testDurationMs);
-
-  function onReceiverChannel(event) {
-    event.channel.addEventListener('message', receive);
-  }
+  function(test) {
+    var wiFiPeriodicScanTest = new WiFiPeriodicScanTest(test);
+    wiFiPeriodicScanTest.run(Call.isRelay);
+  });
 
-  function send() {
-    if (!running) { return; }
-    senderChannel.send('' + Date.now());
-    setTimeout(send, sendIntervalMs);
-  }
+function WiFiPeriodicScanTest(test) {
+  this.test = test;
+  this.testDurationMs = 5 * 60 * 1000;
+  this.sendIntervalMs = 100;
+  this.delays = [];
+  this.recvTimeStamps = [];
+  this.running = false;
+}
+
+WiFiPeriodicScanTest.prototype = {
+  run: function(candidateFilter) {
+    var start = function(candidateFilter, config) {
```

Comment:
make this a prototype function and just do:

Call.asyncCreateTurnConfig(this.start.bind(this), ...);